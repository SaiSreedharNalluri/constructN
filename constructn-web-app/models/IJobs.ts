export interface IJobs {
  _id: string;
  project: Project;
  structure: Structure;
  createdBy: CreatedBy;
  assignedTo: string;
  status: string;
  captures?: CapturesEntityOrSelectedCapturesEntity[];
  selectedCaptures?: CapturesEntityOrSelectedCapturesEntity[];
  date: string;
  createdAt: string;
  updatedAt: string;
}
export interface Project {
  _id: string;
  type: string;
  name: string;
  company: string;
  email: string;
}
export interface Structure {
  _id: string;
  name: string;
  type: string;
  isExterior: boolean;
  project: string;
  parent: string;
  children?: null[] | null;
  location?: null[] | null;
}
export interface CreatedBy {
  _id: string;
  firstName: string;
  lastName: string;
  email: string;
  fullName: string;
}
export interface CapturesEntityOrSelectedCapturesEntity {
  _id: string;
  user: string;
  project: string;
  structure: string;
  captureDateTime: string;
  jobId: string;
  type: string;
  mode: string;
  isMediaUploaded: boolean;
  approvedStatus?: boolean;
  mediaPath: string;
}
