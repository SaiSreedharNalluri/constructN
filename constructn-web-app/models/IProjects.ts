export interface IProjects {
  _id: string;
  type: string;
  name: string;
  company?: null;
  email: string;
  contact: Contact;
  address: Address;
  location?: number[] | null;
  createdAt: string;
  updatedAt: string;
  jobsOpened?: number;
  LastUpdatedOn?: string;
  coverPhoto: string;
}
export interface Contact {
  code: string;
  number: number;
}
export interface Address {
  zipcode: string;
  city: string;
  state: string;
  country: string;
}
