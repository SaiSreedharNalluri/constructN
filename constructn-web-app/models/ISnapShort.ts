export interface ISnapShort {
  _id: string;
  date: string;
  progress: number;
  project: string;
  createdAt: string;
  updatedAt: string;
  __v: number;
  structure: string;
  status: string;
  visualization: [];
}
