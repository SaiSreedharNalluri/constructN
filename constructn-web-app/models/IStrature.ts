export interface IStrature {
  success: boolean;
  result?: IStratureEntity[] | null;
}
export interface IStratureEntity {
  _id: string;
  name: string;
  type: string;
  parent?: null;
  children?: ChildrenEntity[] | null;
}
export interface ChildrenEntity {
  _id: string;
  name: string;
  type: string;
  parent: string;
  children?: ChildrenEntity1[] | null;
}
export interface ChildrenEntity1 {
  _id: string;
  name: string;
  type: string;
  parent: string;
  children?: null[] | null;
}
