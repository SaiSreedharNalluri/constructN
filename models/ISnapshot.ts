import { IActiveReality, IReality } from "./IReality";

export interface ISnapshot {
  _id: string;
  date: string;
  progress: number;
  project: string;
  createdAt: string;
  updatedAt: string;
  __v: number;
  structure: string;
  status: string;
  visualization: [];
  reality?: [IReality];
}

// export interface ISnapShotDeatils {
//   _id: string;
//   progress: number;
//   status: string;
//   visualizations?: null[] | null;
//   structure: StructureOrProject;
//   date: string;
//   project: StructureOrProject;
//   createdAt: string;
//   updatedAt: string;
// }
export interface StructureOrProject {
  _id: string;
  name: string;
}
