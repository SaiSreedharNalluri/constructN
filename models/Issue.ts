import { IContext } from './ITools';

export interface Issue {
  splice(index: number, arg1: number): unknown;
  title: string;
  type: string;
  description: string;
  status: string;
  priority: string;
  screenshot: string;
  assignees: [
    { _id: string; firstName: string; lastName: string; email: string }
  ];
  owner: string;
  project: string;
  structure: string;
  progress: 0;
  tags: [string];
  attachments: [
    {
      name: string;
      url: string;
      entity: string;
      _id: string;
    }
  ];
  createdAt: string;
  updatedAt: string;
  startDate: string;
  dueDate: string;
  context?: IContext;
  _id: string;
  __v: 0;
}
