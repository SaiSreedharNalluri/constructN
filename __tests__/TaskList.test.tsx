import { fireEvent, render, screen } from "@testing-library/react";
import { ISnapshot } from "../models/ISnapshot";
import '@testing-library/jest-dom';
import CustomIssueListDrawer from "../components/divami_components/issue-listing/IssueList";
import { Issue } from "../models/Issue";
import CustomTaskListDrawer from "../components/divami_components/task_list/TaskList";
import TaskList from "../components/divami_components/task_list/TaskList";


const currentSnapshot: ISnapshot = {
    createdAt:"2023-01-24T15:23:40.031Z",
date:"2023-1-06",
progress:-1,
project:"PRJ201897",
reality:[
    {capture:"CAP820581",
    completedDateTime :"2023-01-30T04:55:02.797Z",
    // createdAt :"2023-01-30T04:55:05.466Z",
    job :"JOB820391",
    mode :"Drone Image",
    snapshot   :    "SNP820031",
    status :"active",
    type:"exterior",
    // updatedAt:"2023-01-30T04:55:05.466Z",
    _id :"RLT832872"}
],
status:"Inactive",
structure:"STR536138",
updatedAt:"2023-01-30T04:55:05.466Z",
// visualizations:[]
}
const str={
    "_id": "STR664865",
    "name": "1_Floor_A3_Block",
    "type": "Interior",
    "isExterior": false,
    "project": "PRJ201897",
    "parent": "STR536138",
    "children": [],
    "location": [],
    "designs": [
        {
            "_id": "DSG638729",
            "type": "Plan Drawings",
            "name": "floormap.pdf",
            "project": "PRJ201897",
            "structure": "STR664865",
            "storage": [
                {
                    "provider": "constructn-oss",
                    "path": "PRJ201897/structures/STR664865/designs/DSG638729/floormap.pdf",
                    "providerType": "internal",
                    "format": ".PDF"
                },
                {
                    "provider": "autodesk-oss",
                    "pathId": "dXJuOmFkc2sub2JqZWN0czpvcy5vYmplY3Q6YWRoYW5pX2tvcmVnYW9uX3BhcmtfcHVuZS8lMkZQUkoyMDE4OTclMkZzdHJ1Y3R1cmVzJTJGU1RSNjY0ODY1JTJGZGVzaWducyUyRkRTRzYzODcyOSUyRkEzLUYxLnBkZg",
                    "providerType": "external",
                    "format": ".SVF2"
                }
            ],
            "createdAt": "2023-02-09T12:38:46.704Z",
            "updatedAt": "2023-02-09T12:38:46.704Z",
            "__v": 0,
            "tm": {
                "tm": [
                    4.201579,
                    0.574277,
                    0,
                    385366.46875,
                    -0.574277,
                    4.201579,
                    0,
                    2049775.125,
                    0,
                    0,
                    4.240644,
                    0,
                    0,
                    0,
                    0,
                    1
                ],
                "offset": [
                    0,
                    0,
                    100
                ]
            }
        },
        {
            "_id": "DSG740607",
            "type": "BIM",
            "name": "Block A3.rvt",
            "project": "PRJ201897",
            "structure": "STR664865",
            "storage": [
                {
                    "provider": "constructn-oss",
                    "path": "PRJ201897/structures/STR664865/designs/DSG740607/Block A3.rvt",
                    "providerType": "internal",
                    "format": ".RVT"
                },
                {
                    "provider": "autodesk-oss",
                    "pathId": "dXJuOmFkc2sub2JqZWN0czpvcy5vYmplY3Q6YWRoYW5pX2tvcmVnYW9uX3BhcmtfcHVuZS8lMkZQUkoyMDE4OTclMkZzdHJ1Y3R1cmVzJTJGU1RSNTM2MTM4JTJGZGVzaWducyUyRkRTRzIzNDI5MyUyRmFkaGFuaV9hM19ibG9jay5ydnQ",
                    "providerType": "external",
                    "format": ".SVF2"
                }
            ],
            "createdAt": "2023-02-09T12:38:46.704Z",
            "updatedAt": "2023-02-09T12:38:46.704Z",
            "__v": 0,
            "tm": {
                "tm": [
                    -0.136552,
                    0.992896,
                    0,
                    385382.75,
                    -0.992896,
                    -0.136552,
                    0,
                    2049802,
                    0,
                    0,
                    1.002242,
                    95.249374,
                    0,
                    0,
                    0,
                    1
                ],
                "offset": [
                    385383.538,
                    2049820.99,
                    100
                ],
                "properties_child": [
                    "IfcGUID",
                    "IfcGUID"
                ],
                "properties_head": [
                    "IFC Parameters"
                ]
            }
        }
    ],
    "createdAt": "2023-01-24T10:04:24.865Z",
    "updatedAt": "2023-02-09T12:38:46.704Z",
    "__v": 0
}
const issues = [
    {
      "_id": "TSK557956",
      "title": "task_1",
      "description": "",
      "type": "Transmittals",
      "status": "To Do",
      "priority": "High",
      "assignees": [
        {
          "_id": "USR370060",
          "firstName": "swathi",
          "lastName": "divami",
          "email": "swathi@divami.com",
          "fullName": "swathi divami"
        },
        {
          "_id": "USR370060",
          "firstName": "swathi",
          "lastName": "divami",
          "email": "swathi@divami.com",
          "fullName": "swathi divami"
        }
      ],
      "owner": {
        "_id": "USR370060",
        "firstName": "swathi",
        "lastName": "divami",
        "fullName": "swathi divami"
      },
      "project": "PRJ201897",
      "structure": "STR654406",
      "snapshot": "SNP746698",
      "context": {
        "id": "Temp Task",
        "type": "Task",
        "cameraObject": {
          "cameraPosition": {
            "x": 385387.185919544,
            "y": 2049797.3090976952,
            "z": 159.85971788829193
          },
          "cameraTarget": {
            "x": 385387.185919544,
            "y": 2049797.309151555,
            "z": 106
          }
        },
        "tag": {
          "tagPosition": {
            "x": 385397.5663064057,
            "y": 2049801.8890134313,
            "z": 106
          }
        }
      },
      "progress": -1,
      "sequenceNumber": 422,
      "tags": [],
      "createdAt": "2023-03-02T07:02:37.956Z",
      "updatedAt": "2023-03-02T07:02:38.306Z",
      "screenshot": "https://constructn-attachments-dev.s3.ap-south-1.amazonaws.com/TSK/TSK557956/1677740557665-screenshot.png",
      "attachments": [
        {
          "_id": "ATT558297",
          "name": "Screenshot 2023-02-27 at 2.32.16 PM.png",
          "url": "https://constructn-attachments-dev.s3.ap-south-1.amazonaws.com/TSK/TSK557956/1677740557633-Screenshot%202023-02-27%20at%202.32.16%20PM.png",
          "entity": "TSK557956",
          "createdAt": "2023-03-02T07:02:38.296Z",
          "updatedAt": "2023-03-02T07:02:38.296Z"
        }
      ]
    },
    {
      "_id": "TSK724881",
      "title": "Task",
      "description": "csdc",
      "type": "Submittals",
      "status": "To Do",
      "priority": "Low",
      "assignees": [
        {
          "_id": "USR370060",
          "firstName": "swathi",
          "lastName": "divami",
          "email": "swathi@divami.com",
          "fullName": "swathi divami"
        }
      ],
      "owner": {
        "_id": "USR370060",
        "firstName": "swathi",
        "lastName": "divami",
        "fullName": "swathi divami"
      },
      "project": "PRJ201897",
      "structure": "STR654406",
      "snapshot": "SNP746698",
      "context": {
        "id": "Temp Task",
        "type": "Task",
        "cameraObject": {
          "cameraPosition": {
            "x": 385387.185919544,
            "y": 2049797.3090976952,
            "z": 159.85971788829193
          },
          "cameraTarget": {
            "x": 385387.185919544,
            "y": 2049797.309151555,
            "z": 106
          }
        },
        "tag": {
          "tagPosition": {
            "x": 385381.6360181819,
            "y": 2049799.3878389273,
            "z": 106
          }
        }
      },
      "progress": -1,
      "sequenceNumber": 425,
      "tags": [],
      "createdAt": "2023-03-02T08:12:04.881Z",
      "updatedAt": "2023-03-02T08:12:04.881Z",
      "attachments": []
    },
    {
      "_id": "TSK821452",
      "title": "Task",
      "description": "Testing Assign",
      "type": "Submittals",
      "status": "To Do",
      "priority": "Low",
      "assignees": [
        {
          "_id": "USR370060",
          "firstName": "swathi",
          "lastName": "divami",
          "email": "swathi@divami.com",
          "fullName": "swathi divami"
        }
      ],
      "owner": {
        "_id": "USR370060",
        "firstName": "swathi",
        "lastName": "divami",
        "fullName": "swathi divami"
      },
      "project": "PRJ201897",
      "structure": "STR654406",
      "snapshot": "SNP746698",
      "context": {
        "id": "Temp Task",
        "type": "Task",
        "cameraObject": {
          "cameraPosition": {
            "x": 385387.185919544,
            "y": 2049797.3090976952,
            "z": 159.85971788829193
          },
          "cameraTarget": {
            "x": 385387.185919544,
            "y": 2049797.309151555,
            "z": 106
          }
        },
        "tag": {
          "tagPosition": {
            "x": 385383.08423083014,
            "y": 2049807.1806022697,
            "z": 106
          }
        }
      },
      "progress": -1,
      "sequenceNumber": 426,
      "tags": [],
      "createdAt": "2023-03-02T08:13:41.452Z",
      "updatedAt": "2023-03-02T08:13:41.452Z",
      "attachments": []
    },
    {
      "_id": "TSK975065",
      "title": "Task",
      "description": "description",
      "type": "RFI",
      "status": "To Do",
      "priority": "Low",
      "assignees": [
        {
          "_id": "USR370060",
          "firstName": "swathi",
          "lastName": "divami",
          "email": "swathi@divami.com",
          "fullName": "swathi divami"
        }
      ],
      "owner": {
        "_id": "USR370060",
        "firstName": "swathi",
        "lastName": "divami",
        "fullName": "swathi divami"
      },
      "project": "PRJ201897",
      "structure": "STR654406",
      "snapshot": "SNP746698",
      "context": {
        "id": "Temp Task",
        "type": "Task",
        "cameraObject": {
          "cameraPosition": {
            "x": 385387.185919544,
            "y": 2049797.3090976952,
            "z": 159.85971788829193
          },
          "cameraTarget": {
            "x": 385387.185919544,
            "y": 2049797.309151555,
            "z": 106
          }
        },
        "tag": {
          "tagPosition": {
            "x": 385389.22189395817,
            "y": 2049809.3184400008,
            "z": 106
          }
        }
      },
      "progress": -1,
      "sequenceNumber": 429,
      "tags": [],
      "createdAt": "2023-03-02T09:39:35.064Z",
      "updatedAt": "2023-03-02T09:39:35.064Z",
      "attachments": []
    },
    {
      "_id": "TSK056780",
      "title": "Task",
      "description": "Description ",
      "type": "RFI",
      "status": "To Do",
      "priority": "Low",
      "assignees": [
        {
          "_id": "USR370060",
          "firstName": "swathi",
          "lastName": "divami",
          "email": "swathi@divami.com",
          "fullName": "swathi divami"
        }
      ],
      "owner": {
        "_id": "USR370060",
        "firstName": "swathi",
        "lastName": "divami",
        "fullName": "swathi divami"
      },
      "project": "PRJ201897",
      "structure": "STR654406",
      "snapshot": "SNP746698",
      "context": {
        "id": "Temp Task",
        "type": "Task",
        "cameraObject": {
          "cameraPosition": {
            "x": 385387.185919544,
            "y": 2049797.3090976952,
            "z": 159.85971788829193
          },
          "cameraTarget": {
            "x": 385387.185919544,
            "y": 2049797.309151555,
            "z": 106
          }
        },
        "tag": {
          "tagPosition": {
            "x": 385392.1872817616,
            "y": 2049806.7669329506,
            "z": 106
          }
        }
      },
      "progress": -1,
      "sequenceNumber": 430,
      "tags": [],
      "createdAt": "2023-03-02T09:40:56.779Z",
      "updatedAt": "2023-03-02T09:40:56.779Z",
      "attachments": []
    }
  ]
jest.mock('next/router', () => ({
            route: '/',
            pathname: '',
            query: { projectId: '1234' },
            isReady:true,
            asPath: '',
            push: jest.fn(),
            events: {
              on: jest.fn(),
              off: jest.fn()
            },
            beforePopState: jest.fn(() => null),
            prefetch: jest.fn(() => null)
      }));

  
    
    it("renders component with empty issue list", () => {    
        const { container } = 
        render(<TaskList
            tasksList={issues}
            taskMenuClicked={jest.fn()}
            currentProject={"PRJ201897"}
            currentStructure={str}
            currentSnapshot={currentSnapshot}
            contextInfo={{type: 'Task'}}
            closeTaskFilterOverlay={jest.fn()}
            handleOnTaskFilter={jest.fn()}
            onClose={jest.fn()}
            deleteTheTask={jest.fn()}
            taskFilterState={{
                isFilterApplied: false,
                filterData: {},
                numberOfFilters: 0,
              }}
            getTasks={jest.fn()}
            handleOnTasksSort={jest.fn()}
            deleteTheAttachment={jest.fn()}
          />)
        expect(container).toMatchSnapshot();
        expect(screen.getByText("Task List")).toBeInTheDocument();
            const s = screen.getByTestId('search-icon');
            fireEvent.click(s);
            const search = screen.getAllByPlaceholderText('Search');
            fireEvent.change(search[0],{target:{value:'A'}});
            expect(search[0]).toBeInTheDocument();
            fireEvent.click(screen.getByTestId("search-close"))
            fireEvent.click(screen.getAllByTestId('item-body')[0])
            const closeIcon = screen.getByTestId('close-icon')
            fireEvent.click(closeIcon);
            // expect(screen.getByText("Task List")).not.toBeInTheDocument();
    });
    it("renders component with empty issue list", () => {    
        const { container } = 
        render(<TaskList
            tasksList={[]}
            taskMenuClicked={jest.fn()}
            currentProject={"PRJ201897"}
            currentStructure={str}
            currentSnapshot={currentSnapshot}
            contextInfo={{type: 'Task'}}
            closeTaskFilterOverlay={jest.fn()}
            handleOnTaskFilter={jest.fn()}
            onClose={jest.fn()}
            deleteTheTask={jest.fn()}
            taskFilterState={{
                isFilterApplied: true,
                filterData: {},
                numberOfFilters: 0,
              }}
            getTasks={jest.fn()}
            handleOnTasksSort={jest.fn()}
            deleteTheAttachment={jest.fn()}
          />)
        expect(container).toMatchSnapshot();
        
    });
    it('renders component with data',() => {
        const { container } = render(<TaskList
            tasksList={issues}
            taskMenuClicked={jest.fn()}
            currentProject={"PRJ201897"}
            currentStructure={str}
            currentSnapshot={currentSnapshot}
            contextInfo={{type: 'Task'}}
            closeTaskFilterOverlay={jest.fn()}
            handleOnTaskFilter={jest.fn()}
            onClose={jest.fn()}
            deleteTheTask={jest.fn()}
            taskFilterState={{
                isFilterApplied: false,
                filterData: {},
                numberOfFilters: 0,
              }}
            getTasks={jest.fn()}
            handleOnTasksSort={jest.fn()}
            deleteTheAttachment={jest.fn()}
          />)
          const sort = screen.getByTestId("sort")
          fireEvent.click(sort);
          expect(screen.getByText("Status ( To Do - Completed)")).toBeInTheDocument()
          const sortitem = screen.getAllByTestId("sort-menu-item");
          fireEvent.click(sortitem[0])
          const download = screen.getByTestId("download");
          fireEvent.click(download)
          
          const filter = screen.getByTestId("filter");
          fireEvent.click(filter)
          expect(screen.getByText("Filters")).toBeInTheDocument()
          fireEvent.click(screen.getByTestId("filter-close"))
          expect(screen.getByText("Task List")).toBeInTheDocument()
    });

    // it('check on click functionalities',() => {
    //     render(
    //         <CustomTaskListDrawer
    //                 closeFilterOverlay={()=>{}}
    //                 issuesList={issues as unknown as Issue[]}
    //                 visibility={false}
    //                 closeOverlay={jest.fn()}
    //                 handleOnFilter={jest.fn()}
    //                 onClose={jest.fn()}
    //                 handleOnSort={jest.fn()}
    //                 deleteTheIssue={()=>{}}
    //                 clickIssueEditSubmit={jest.fn()}
    //                 issuePriorityList={ ['High', 'Low', 'Medium']}
    //                 issueStatusList={['Blocked', 'To Do', 'In Progress', 'Completed']}
    //                 currentStructure={str}
    //                 currentSnapshot={currentSnapshot}
    //                 contextInfo={
    //                     {type: 'Task'}}
    //                 currentProject={'"PRJ201897"'}
    //                 issueTypesList={['Clash', 'Building code', 'Design', 'Safety', 'Commissioning']}
    //                 issueFilterState={{filterData:{},
    //                     isFilterApplied :false,
    //                     numberOfFilters :0}}
    //                 getIssues={()=>{}}
    //                 handleOnIssueSort={()=>{}}
    //                 deleteTheAttachment={()=>{}}
    //             />
    //         );
    //         const sort = screen.getByTestId("sort")
    //         fireEvent.click(sort);
    //         expect(screen.getByText("Status ( To Do - Completed)")).toBeInTheDocument()
    //         const sortitem = screen.getAllByTestId("sort-menu-item");
    //         fireEvent.click(sortitem[0])
    //         const download = screen.getByTestId("download");
    //         fireEvent.click(download)
            
    //         const filter = screen.getByTestId("filter");
    //         fireEvent.click(filter)
    //         expect(screen.getByText("Filters")).toBeInTheDocument()
    //         fireEvent.click(screen.getByTestId("filter-close"))
    //         expect(screen.getByText("Issue List")).toBeInTheDocument()

    // })