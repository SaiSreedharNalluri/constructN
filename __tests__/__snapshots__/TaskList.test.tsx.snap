// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component with empty issue list 1`] = `
<div>
  <div
    class="css-e7sl8t"
  >
    <div
      class="css-mph6fm"
    >
      <div
        class="MuiBox-root css-h9ywjr"
      >
        <span>
          Task List
        </span>
        <img
          alt="close icon"
          class="css-1op8d0u"
          data-nimg="1"
          data-testid="close-icon"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-ev3z9q"
    >
      <div
        class="MuiBox-root css-s58x5k"
      >
        <img
          alt="close icon"
          class="css-4g6ai3"
          data-nimg="1"
          data-testid="search-icon"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt=""
          class="css-65i07b"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Arrow"
          aria-label="Sort Menu"
          class=" css-pswt6o"
          data-mui-internal-clone-element="true"
          data-nimg="1"
          data-testid="sort"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt=""
          class="css-1yesrfp"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Arrow"
          class="css-pswt6o"
          data-nimg="1"
          data-testid="filter"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <a
          class="text-black btn btn-primary fill-black fa fa-Download "
          data-testid="download"
          download="my-tasks.csv"
          href="data:text/csv;charset=utf-8,ï»¿"_id","title","description","type","status","priority","assignees","owner","project","structure","snapshot","sequenceNumber","tags","createdAt","updatedAt","screenshot","attachments","assignee"
"TSK557956","task_1","","Transmittals","To Do","High","[object Object],[object Object]","[object Object]","PRJ201897","STR654406","SNP746698","422","","2023-03-02T07:02:37.956Z","2023-03-02T07:02:38.306Z","https://constructn-attachments-dev.s3.ap-south-1.amazonaws.com/TSK/TSK557956/1677740557665-screenshot.png","[object Object]",""
"TSK724881","Task","csdc","Submittals","To Do","Low","[object Object]","[object Object]","PRJ201897","STR654406","SNP746698","425","","2023-03-02T08:12:04.881Z","2023-03-02T08:12:04.881Z","","",""
"TSK821452","Task","Testing Assign","Submittals","To Do","Low","[object Object]","[object Object]","PRJ201897","STR654406","SNP746698","426","","2023-03-02T08:13:41.452Z","2023-03-02T08:13:41.452Z","","",""
"TSK975065","Task","description","RFI","To Do","Low","[object Object]","[object Object]","PRJ201897","STR654406","SNP746698","429","","2023-03-02T09:39:35.064Z","2023-03-02T09:39:35.064Z","","",""
"TSK056780","Task","Description ","RFI","To Do","Low","[object Object]","[object Object]","PRJ201897","STR654406","SNP746698","430","","2023-03-02T09:40:56.779Z","2023-03-02T09:40:56.779Z","","","""
          target="_self"
        >
          <img
            alt="Arrow"
            class="css-8k5w76"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiBox-root css-1t2b8fl"
    >
      <div
        class="MuiBox-root css-w6lgmz"
      >
        <div
          class="css-4g6ai3"
          data-testid="item-body"
        >
          <div
            class="css-k008qs"
          >
            <img
              alt="Arr"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="css-1y6ic72"
            >
              Transmittals
               (#
              422
              )
            </div>
          </div>
          <div
            class="css-17nntv"
          >
            <div>
              High
               Priority
            </div>
          </div>
          <div
            class="css-joyevq"
          >
            <div />
            <div
              class="css-lahlre"
            >
              Due by 
              03 Mar '23
            </div>
          </div>
        </div>
        <div
          class="css-1vabvql"
        />
        <div
          class="css-4g6ai3"
          data-testid="item-body"
        >
          <div
            class="css-k008qs"
          >
            <img
              alt="Arr"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="css-1y6ic72"
            >
              Submittals
               (#
              425
              )
            </div>
          </div>
          <div
            class="css-17nntv"
          >
            <div>
              Low
               Priority
            </div>
          </div>
          <div
            class="css-joyevq"
          >
            <div />
            <div
              class="css-lahlre"
            >
              Due by 
              03 Mar '23
            </div>
          </div>
        </div>
        <div
          class="css-1vabvql"
        />
        <div
          class="css-4g6ai3"
          data-testid="item-body"
        >
          <div
            class="css-k008qs"
          >
            <img
              alt="Arr"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="css-1y6ic72"
            >
              Submittals
               (#
              426
              )
            </div>
          </div>
          <div
            class="css-17nntv"
          >
            <div>
              Low
               Priority
            </div>
          </div>
          <div
            class="css-joyevq"
          >
            <div />
            <div
              class="css-lahlre"
            >
              Due by 
              03 Mar '23
            </div>
          </div>
        </div>
        <div
          class="css-1vabvql"
        />
        <div
          class="css-4g6ai3"
          data-testid="item-body"
        >
          <div
            class="css-k008qs"
          >
            <img
              alt="Arr"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="css-1y6ic72"
            >
              RFI
               (#
              429
              )
            </div>
          </div>
          <div
            class="css-17nntv"
          >
            <div>
              Low
               Priority
            </div>
          </div>
          <div
            class="css-joyevq"
          >
            <div />
            <div
              class="css-lahlre"
            >
              Due by 
              03 Mar '23
            </div>
          </div>
        </div>
        <div
          class="css-1vabvql"
        />
        <div
          class="css-4g6ai3"
          data-testid="item-body"
        >
          <div
            class="css-k008qs"
          >
            <img
              alt="Arr"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="css-1y6ic72"
            >
              RFI
               (#
              430
              )
            </div>
          </div>
          <div
            class="css-17nntv"
          >
            <div>
              Low
               Priority
            </div>
          </div>
          <div
            class="css-joyevq"
          >
            <div />
            <div
              class="css-lahlre"
            >
              Due by 
              03 Mar '23
            </div>
          </div>
        </div>
        <div
          class="css-1vabvql"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders component with empty issue list 2`] = `
<div>
  <div
    class="css-e7sl8t"
  >
    <div
      class="css-1tfdt9d"
    >
      <img
        alt="Error Image"
        class="css-hctw80"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <div
        class="css-1rqdy21"
      >
        No Task has been raised yet. Get a headstart by raising one.
      </div>
      <div
        class="css-1tivz4q"
      >
        Raise Task
      </div>
      <div
        class="css-ss5hb"
      >
        Check out
        <span
          class="css-dbdm06"
        >
           How to raise a Task?
        </span>
      </div>
    </div>
  </div>
</div>
`;
